tsdb/compact.go

```diff
@@ -730,107 +723,58 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 		return errors.Wrap(symbols.Err(), "next symbol")
 	}
 
-	delIter := &deletedIterator{}
-	ref := uint64(0)
+	var (
+		ref  = uint64(0)
+		chks []chunks.Meta
+	)
+
+	set := sets[0]
+	if len(sets) > 1 {
+		// Merge series using compacting chunk series merger.
+		set = storage.NewMergeChunkSeriesSet(sets, storage.NewCompactingChunkSeriesMerger(storage.ChainedSeriesMerge))
+	}
+
+	// Iterate over all sorted chunk series.
 	for set.Next() {
 		select {
 		case <-c.ctx.Done():
 			return c.ctx.Err()
 		default:
 		}
 
-		lset, chks, dranges := set.At() // The chunks here are not fully deleted.
-		if overlapping {
-			// If blocks are overlapping, it is possible to have unsorted chunks.
-			sort.Slice(chks, func(i, j int) bool {
-				return chks[i].MinTime < chks[j].MinTime
-			})
+		chksIter := set.At().Iterator()
+		chks = chks[:0]
+		for chksIter.Next() {
+			// TODO(bwplotk): We now iterate over chunks, consider using this fact to stream writes more.
```

Comment:
I don't think that'd be wise. We're already doing effectively one write per series, and even a 1s interval series is only going to be around 2.5MB. Smaller would mean more syscalls, and likely end up slower.

Reply:
> We're already doing effectively one write per series

What do you mean "one write" here in context of fetching series?

Reply:
ok, ignore me it's compact ok. This is fair then.

Reply:
Adding comment:

```
	// We are not iterating in streaming way over chunk as it's more efficient to do bulk write for index and
			// chunk file purposes.
```